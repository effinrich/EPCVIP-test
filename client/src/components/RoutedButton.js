import React from 'react'
import PropTypes from 'prop-types'
import { matchPath, withRouter } from 'react-router'
import { Button } from 'grommet'

// A simple component that shows the pathname of the current location
const RoutedButton = ({
  active,
  exact,
  match,
  location,
  history,
  path,
  strict,
  onClick,
  ...rest
}) => {
  const pathMatch = matchPath(location.pathname, { exact, path, strict })

  const handleOnClick = event => {
    event.preventDefault()

    if (onClick) {
      onClick(event)
    }

    history.push(path)
    event.stopPropagation()
  }

  return (
    <Button active={active && !!pathMatch} onClick={handleOnClick} {...rest} />
  )
}

RoutedButton.propTypes = {
  location: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  path: PropTypes.string,
  active: PropTypes.bool,
  exact: PropTypes.bool,
  match: PropTypes.object,
  strict: PropTypes.bool,
  onClick: PropTypes.func
}

export default withRouter(RoutedButton)
