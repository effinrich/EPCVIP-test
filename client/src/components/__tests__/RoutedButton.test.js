import React from 'react'

import { render } from 'utils/testUtils/helperRtl'

import RoutedButton from '../RoutedButton'

describe('RoutedButton Component', () => {
  it('should render without crashing', () => {
    const { container } = render(<RoutedButton />)
    expect(container).toMatchSnapshot()
  })

  it('should render child element', () => {
    const { container } = render(<RoutedButton />)
    expect(container.firstChild).toBeDefined()
  })
})
