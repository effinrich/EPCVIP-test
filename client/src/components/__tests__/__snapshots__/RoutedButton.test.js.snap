// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoutedButton Component should render without crashing 1`] = `
<div>
  <button
    class="StyledButton-sc-323bzc-0 iZFCLM"
    type="button"
  />
</div>
`;
